### PR TITLE
Add socks5 proxy via dante

### PR DESCRIPTION
**Pull request by Fraggle**
## Description

This PR adds a SOCKS5 proxy using Dante. The changes include:

1. Adding a new Dante Dockerfile and configuration
2. Creating a new GitHub Actions workflow for deploying Dante
3. Updating Kubernetes configurations to include the SOCKS proxy
4. Minor adjustments to existing configurations

Usage:
Use the SOCKS_PROXY_HOST and SOCKS_PROXY_PORT env variables in `core` or `connectors` (already available).

## Risk

None

## Deploy Plan

Apply infra

### CODE REVIEW

**Comment by spolu** [comment]
I'm a bit lost between nginx and dante and who uses what when and where. Do we have a design doc that describes the latest state?

File: dante.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,24 @@
+FROM alpine:3.20
```
Comment (fontanierh) [inline dante.Dockerfile:1]: What is special about our image? Ideally if we coule use a stock one we wouldn't need a github workflow etc...

File: dante.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    && cd / \ 
+    && adduser -S -D -u 8062 -H sockd \
+    && curl -Lo /usr/local/bin/dumb-init https://github.com/Yelp/dumb-init/releases/download/v1.1.3/dumb-init_1.1.3_amd64 \
+    && chmod +x /usr/local/bin/dumb-init \
+    && rm -rf /tmp/* \
+    && apk del --purge .build-deps
+
+EXPOSE 1080
+
+ENTRYPOINT ["dumb-init"]
```
Comment (fontanierh) [inline dante.Dockerfile:22]: What does it do?

**Comment by Fraggle** [comment]
Nginx = http proxy
Dante = socks proxy

Norhing uses dante yet.

The design doc is not up to date, i’ll update (btw i understood than design doc were only a point in time and they were not kept updated, is that incorrect ?)

File: dante.Dockerfile (hunk 2 above)
Comment (Fraggle) [reply dante.Dockerfile:22]: Avoid the container to immediately shutdown.

File: dante.Dockerfile (hunk 1 above)
Comment (Fraggle) [reply dante.Dockerfile:1]: The one i used before was also very old and the creator was using « ash » (making it different from our others images, breaking k8s_ssh for example)

I’m not a fan of the extra workflow either but we shouldn’t have to run it often.

Comment (fontanierh) [reply dante.Dockerfile:1]: Ok if there's no official image then LGTM 👍

**Review by fontanierh — APPROVED** [review]
LGTM

**Comment by spolu** [comment]
> The design doc is not up to date, i’ll update (btw i understood than design doc were only a point in time and they were not kept updated, is that incorrect ?)

Well if the project is still ongoing it's good hygiene to have an up to date version so that it can be used to track progress and review to better understand PRs such as here.

re nginx vs dante: I though it was possible to do with nginx? is it? Right now snowflake uses nginx then? what's the plan for dante?

**Comment by Fraggle** [comment]
> > The design doc is not up to date, i’ll update (btw i understood than design doc were only a point in time and they were not kept updated, is that incorrect ?)
> 
> Well if the project is still ongoing it's good hygiene to have an up to date version so that it can be used to track progress and review to better understand PRs such as here.

No problemo, will do

> re nginx vs dante: I though it was possible to do with nginx? is it? 

It is but turned out more hacky than i was expected (running a client in front of nginx).
After discussing quickly live with @fontanierh we decided that it was simpler to use dante.

> Right now snowflake uses nginx then? 

Yes

> What's the plan for dante?

Immediately, nothing. Mostly about readiness for other db engines in « 48h » (such as postgres)

**Comment by spolu** [comment]
ack thanks

**Pull request merged**